homeassistant/components/vera/config_flow.py

```diff
@@ -87,6 +79,9 @@ class VeraFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
     @callback
     def async_get_options_flow(config_entry) -> OptionsFlowHandler:
         """Get the options flow."""
+        if config_entry.data.get(CONF_SOURCE) == config_entries.SOURCE_IMPORT:
+            raise data_entry_flow.UnknownHandler
```

Comment:
We shouldn't try to come up with a way to not allow users to edit imported entries. Either fully support both GUI and yaml options or pick one or the other. Eg it's ok to only allow some options via config yaml.

Reply:
If that's the case, then those options would be overwritten by the values in configuration.yml on every start.

Reply:
If we only allow some options via config yaml, the user can only change them via config yaml so overwriting them on every restart is fine.

Reply:
The options available in the UI and configuration file are the same. The only reason I'm keeping config file support is for backwards comparability. All future integrations should be done through the UI.

Reply:
To support both the best way is this:

- Config yaml options is stored in config entry data at the end of the config flow for new entries.
- If the same entry, checked via unique_id, is imported from config yaml into config flow, we check if config yaml is the same as entry data. If it's the same we just abort. If the new data has changed compared to old data we update both entry data and entry options with the new config yaml data and then abort.
- Default options are set in entry options in `async_setyp_entry` if no options are set yet. The default options data is taken from entry data if it exist there due to config yaml imported entry, otherwise the defaults of the config schema.